File: apis/src/main/java/com/beat/apis/performance/application/dto/performanceDetail/PerformanceDetailImageResponse.java
package com.beat.apis.performance.application.dto.performanceDetail;

import com.beat.global.support.jackson.CdnImageUrl;

public record PerformanceDetailImageResponse(
	Long performanceImageId,
	@CdnImageUrl String performanceImage
) {
	public static PerformanceDetailImageResponse of(Long performanceImageId, String performanceImage) {
		return new PerformanceDetailImageResponse(performanceImageId, performanceImage);
	}
}

File: apis/src/main/kotlin/com/beat/apis/performance/api/response/PerformanceDetailImageResponse.kt
package com.beat.apis.performance.api.response

import com.beat.application.frontoffice.performance.PerformanceImageResult
import com.beat.global.support.jackson.CdnImageUrl

@ConsistentCopyVisibility
data class PerformanceDetailImageResponse private constructor(
    val performanceImageId: Long?, @field:CdnImageUrl val performanceImage: String?,
) {
    companion object {
        fun from(result: PerformanceImageResult): PerformanceDetailImageResponse =
            PerformanceDetailImageResponse(result.id, result.image)
    }
}
